lib/layers/presentation/sing_in_page/widgets/animated_background.dart

```diff
@@ -0,0 +1,47 @@
+import 'package:flutter/material.dart';
+
+class AnimatedBackground extends StatefulWidget {
+  const AnimatedBackground({Key? key}) : super(key: key);
+
+  @override
+  State<AnimatedBackground> createState() => _AnimatedBackgroundState();
+}
+
+class _AnimatedBackgroundState extends State<AnimatedBackground> {
+  Color _newColor = Colors.red.shade200;
+
+  @override
+  Widget build(BuildContext context) {
+    return Center(
+      child: TweenAnimationBuilder(
+        tween: ColorTween(begin: Colors.white, end: _newColor),
+        duration: const Duration(seconds: 3),
+        onEnd: () {
+          setState(() {
+            _newColor = _newColor == Colors.red.shade200
+                ? Colors.white
+                : Colors.red.shade200;
+          });
+        },
+        builder: (_, Color? color, child) {
+          return ColorFiltered(
+            child: MediaQuery.of(context).orientation == Orientation.portrait
```

Comment:
As I see only difference is in url. everything is the same soyou can use ternary operator for url only